core/src/main/java/org/bitcoinj/core/DefaultAddressParser.java

```diff
@@ -16,16 +16,87 @@
 
 package org.bitcoinj.core;
 
+import org.bitcoinj.base.Base58;
+import org.bitcoinj.base.Bech32;
+import org.bitcoinj.base.BitcoinNetwork;
 import org.bitcoinj.base.Network;
 import org.bitcoinj.base.exceptions.AddressFormatException;
+import org.bitcoinj.base.utils.StreamUtils;
+import org.bitcoinj.params.Networks;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
 
 /**
- * Address Parser that knows about the address types supported by bitcoinj core.
+ * Address Parser that knows about the address types supported by bitcoinj core and is configurable
+ * with additional Network types.
  */
 public class DefaultAddressParser implements AddressParser {
+
+    // Networks to try when parsing segwit addresses
+    public static final List<Network> DEFAULT_SEGWIT_NETWORKS = unmodifiableList(
+                                                                    BitcoinNetwork.MAINNET,
+                                                                    BitcoinNetwork.TESTNET,
+                                                                    BitcoinNetwork.REGTEST);
+
+    // Networks to try when parsing base58 addresses
+    public static final List<Network> DEFAULT_BASE58_NETWORKS = unmodifiableList(
```

Comment:
Perhaps just a nitpick, but elsewhere we use the term "legacy" if we're talkabout about pre-segwit-addresses. E.g. it`s `LegacyAddress` rather than Base58Address.

(multiple occurances in member names and comments)

Reply:
Yeah, I thought about using Legacy here for consistency, but I actually think Base58 is a little more descriptive -- especially in the context of parsing. I also thought that `DEFAULT_LEGACY_NETWORKS` would be confusing because they are not legacy networks, but legacy addresses. I suppose we could make it `DEFAULT_NETWORKS_LEGACY_ADDR` with a similar change for segwit.

Reply:
The two issues are orthogonal.

- The naming weirdness you pointed out applies to `DEFAULT_SEGWIT_NETWORKS` as well – there are not segwit networks. Feel free to change the names according to your suggestion, though I think already `DEFAULT_NETWORKS_LEGACY` is less confusing.
- I prefer consistency, and up to now the term "legacy" is pretty established. We can migrate to something different in a later PR, which should imho then rename every occurence of "legacy".

Reply:
OK, I will rename these as follows:

`DEFAULT_SEGWIT_NETWORKS` -> `DEFAULT_NETWORKS_SEGWIT`

`DEFAULT_BASE58_NETWORKS` -> `DEFAULT_NETWORKS_LEGACY`

Reply:
Fixed in f60e4daf4117294fdde24dc4622600b3633d7977